### ISSUE TITLE
Docker VMM not able to bind to external USB volumes

### ISSUE THREAD

**Issue by twoloavesofbread** [docker/for-mac#7480]
### Description

M1 Mac Mini, MacOS Sonoma 14.7.1 
Docker Desktop 4.35.1 (173168)
Docker Engine 27.3.1

***

Switching from Apple Virtualization Framework to Docker VMM causes all containers with bind mounts to external USB volumes to fail with the same error:

`(HTTP code 400) unexpected - invalid mount config for type "bind": bind source path does not exist: /host_mnt/Externalvolumename/sourcefolder` (Docker seems to add `/host_mnt` to the beginning of the file path when it detects an external volume.)

* Bind mounts are added during creation in terminal via `--mount type=bind,source=/sourcefolder,target=/targetvolume`. These folders exist, are not empty, and function on Apple Virtualization Framework.
* Switching back to AVF resolves the issue.
* Running/creating a new container connected to the external bind mounts does **not** resolve the issue. 
* Issue does not occur on bind mounts made to local storage (e.g. `/Users/username/Music` would work for a Navidrome container, for instance).
* When already using Docker VMM, attempting to create/run a container with a bind mount connected to an external volume results in the same error. The container is then **not** run/created.

### Reproduce

For an existing container:
 
1. Ensure Docker is set to use Apple Virtualization Framework (Settings > General > Virtual Machine Options).
2. Run a container in Docker which uses a bind mount tied to a location on an external USB volume (thumb drive, ext hdd, etc).
3. Switch to Docker VMM (Settings > General > Virtual Machine Options).
4. Container will have exited with code 127.
5. Attempt to start container. Observer HTTP 400 error.

***

For a new container:

1. Set VM to Docker VMM
2. Attempt to run/create a container which uses a bind mount tied to a location on an external USB volume (thumb drive, ext hdd, etc).
3. Observe HTTP 400 error.

### Expected behavior

Bind mounts should connect even when they are to an external USB device. Providing a legitimate path to the volume should not cause Docker VMM to not see the volume just because it isn't the primary MacOS HDD.

### docker version

```bash
Client:
 Version:           27.3.1
 API version:       1.47
 Go version:        go1.22.7
 Git commit:        ce12230
 Built:             Fri Sep 20 11:38:18 2024
 OS/Arch:           darwin/arm64
 Context:           desktop-linux

Server: Docker Desktop 4.35.1 (173168)
 Engine:
  Version:          27.3.1
  API version:      1.47 (minimum version 1.24)
  Go version:       go1.22.7
  Git commit:       41ca978
  Built:            Fri Sep 20 11:41:19 2024
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.7.21
  GitCommit:        472731909fa34bd7bc9c087e4c27943f9835f111
 runc:
  Version:          1.1.13
  GitCommit:        v1.1.13-0-g58aa920
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client:
 Version:    27.3.1
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.17.1-desktop.1
    Path:     /Users/jay/.docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.29.7-desktop.1
    Path:     /Users/jay/.docker/cli-plugins/docker-compose
  debug: Get a shell into any image or container (Docker Inc.)
    Version:  0.0.37
    Path:     /Users/jay/.docker/cli-plugins/docker-debug
  desktop: Docker Desktop commands (Alpha) (Docker Inc.)
    Version:  v0.0.15
    Path:     /Users/jay/.docker/cli-plugins/docker-desktop
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.2
    Path:     /Users/jay/.docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.27
    Path:     /Users/jay/.docker/cli-plugins/docker-extension
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  v1.0.5
    Path:     /Users/jay/.docker/cli-plugins/docker-feedback
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v1.3.0
    Path:     /Users/jay/.docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /Users/jay/.docker/cli-plugins/docker-sbom
  scout: Docker Scout (Docker Inc.)
    Version:  v1.14.0
    Path:     /Users/jay/.docker/cli-plugins/docker-scout

Server:
 Containers: 3
  Running: 0
  Paused: 0
  Stopped: 3
 Images: 4
 Server Version: 27.3.1
 Storage Driver: overlayfs
  driver-type: io.containerd.snapshotter.v1
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 472731909fa34bd7bc9c087e4c27943f9835f111
 runc version: v1.1.13-0-g58aa920
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 6.10.11-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: aarch64
 CPUs: 8
 Total Memory: 3.827GiB
 Name: docker-desktop
 ID: 6a131993-21ba-4f66-9dd7-80cf33185c32
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Labels:
  com.docker.desktop.address=unix:///Users/jay/Library/Containers/com.docker.docker/Data/docker-cli.sock
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: daemon is not using the default seccomp profile
```

### Diagnostics ID

7C248197-12EB-4D2D-85AB-0944ED8732CA/20241110001347

**Comment by AldasG** [comment]
Tested with 4.37, issue is still present. Once I swap to Apple Virtualization framework, all containers come up.

However, I have a different error it seems:
(HTTP code 500) server error - error while creating mount source path '/host_mnt/Volumes/Movies/Movies - Dange/Radarr': mkdir /host_mnt/Volumes/Movies/Movies - Dange: read-only file system

**Comment by joreilly02** [comment]
Same thing is happening to me. When I switch back to Apple Virtualization Framework, it works correctly. Can post logs if that will help.

**Comment by enzoxic** [comment]
Yes please be try to make a didacticiel for the newbees, if it’s not to much asking?

Envoyé à partir de Outlook pour iOS<https://aka.ms/o0ukef>
________________________________
De : joreilly02 ***@***.***>
Envoyé : Tuesday, December 31, 2024 5:06:51 PM
À : docker/for-mac ***@***.***>
Cc : Subscribed ***@***.***>
Objet : Re: [docker/for-mac] Docker VMM not able to bind to external USB volumes (Issue #7480)


Same thing is happening to me. When I switch back to Apple Virtualization Framework, it works correctly. Can post logs if that will help.

**Comment by Utesgui** [comment]
i have kind of the same issue. I switched from Apple Virtualization Framework to Docker VMM and my containers couldn't start, because it says permission denied ...
![Image](https://github.com/user-attachments/assets/c0c92c33-9d86-405e-b4f8-8ee4eb1c43ff)

I use a 2TB NVME disk, connected via Thunderbolt 4 (USB-C) that is mounted to /Volumes/arr-cache

Mac mini M4
Docker Desktop 4.37.2
Docker Engine 27.4.0

All Soft-/Firmware is up-to-date at time of writing.

Restarting the Docker Engine or the whole Mac does not help.
Switching back to Apple Virtualization Framework and my containers start up just fine.

**Comment by jt3204** [comment]
Same issue. When accessing the VM directly and doing ls -l in the Volumes folder, the shared mount appears with question marks like this:

lrwxr-xr-x  1 root root     1 Aug 18 06:46 'Macintosh HD' -> /
drwx------  1 root root 16384 Jul 27 13:11 'Shared Drive'
drwxrwxr-x  4 root root   128 Aug 18 06:49 "BDD"
drwxrwxr-x 13 root root   510 Aug 18 03:56 "FDD"
drwxr-xr-x  9 root root   288 Mar 15 23:53  Backup.tibx
d?????????  ? ?    ?        ?            ?  VDD

Interesting is that the Shared Drive is also a shared mount, but displays correctly until after trying to access it (which errors "no such file or directory".

**Comment by jordanlambrecht** [comment]
Also having this issue. Has the docker team acknowledged this? It's a pretty huge deal and kind of wild that it's still ongoing for over a year.

**Comment by ChristofferNissen** [comment]
Can report same error happens when trying to bind a folder on an SMB share

**Comment by Necriso** [comment]
Still exists on
Docker Desktop: 4.91.0 (239619)
Engine: 29.8.0

will this ever be fixed?
